index.js

```diff
@@ -103,8 +104,9 @@ export default class Carousel extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.children !== nextProps.children) {
+    if (!isEqual(this.props.children, nextProps.children)) {
       let childrenLength = 0;
+      this.setState({ currentPage: 1 });
```

Comment:
Hm.. Maybe lets check children length and if it's lower then the `currentPage` set it to 1 ?